frontend/src/pages/Account.tsx

```diff
@@ -1,37 +1,51 @@
-import React, { useMemo, useState } from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import useAsyncEffect from 'use-async-effect';
 import { useParams } from 'react-router-dom';
 import { useTranslation } from 'react-i18next';
-
-import { useAccount } from 'src/hooks';
+import {
+  fetchAccount,
+  Loading,
+  useAppDispatch,
+  useAppSelector,
+} from 'src/store';
+import {
+  getAccount,
+  getAccountErrorMessage,
+  getAccountLoadingStatus,
+} from 'src/store/selectors/account-selectors';
 import { casperApi } from '../api';
 import { AccountDetailsCard, PageHead, PageWrapper } from '../components';
 
 export const AccountPage: React.FC = () => {
   const { id } = useParams();
   const { t } = useTranslation();
-  const {
-    data: account,
-    error: accountError,
-    isLoading,
-  } = useAccount({
-    accountHashOrPublicKey: id || '',
-  });
+
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    dispatch(fetchAccount(id ?? ''));
+  }, [dispatch, id]);
+
+  const account = useAppSelector(getAccount);
+  const accountLoadingStatus = useAppSelector(getAccountLoadingStatus);
+  const accountErrorMessage = useAppSelector(getAccountErrorMessage);
+
+  const isLoading = accountLoadingStatus !== Loading.Complete;
 
   const [balance, setBalance] = useState<string | null>(null);
 
   useAsyncEffect(async () => {
     if (account) {
+      // TODO: move to middleware once repo has been ejected
```

Comment:
This will be moved over to the middleware once the FE/BE is split, and the data will be fetched using the standard redux/API patterns.

Reply:
yeah we should move it but do not cache the value in the middleware. and then we can 100% get rid of `casperApi` :) 

[Middleware ticket](https://github.com/casper-network/casper-blockexplorer-middleware/issues/11)
[Frontend ticket](https://github.com/casper-network/casper-blockexplorer/issues/183)